Automattic/jetpack
# Search: remove unexpected sidebar widgets added on theme switch

Partly fixes #18556

#### Changes proposed in this Pull Request:
Move widgets added by WP to inactive widgets when new theme has less sidebars than the old theme, by hooking `after_switch_theme` to save a copy of old sidebar setting and hooking `pre_update_option_sidebars_widgets` to do the sweeping.

Another possible solution is to change the instant search sidebar name from `jetpack-instant-search-sidebar` to a name without keyword `sidebar`. The hack would walk around the widgets migrations strategy by WordPress. The problem with its compatibility, which will result in some code migrating the old setting to the new one otherwise search sidebar settings would be lost.

#### Jetpack product discussion
pcNPJE-96-p2

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Go to a site with Jetpack Search subscription
- Switch theme to `Twenty Twenty` which has two sidebar area ("Footer #1" and "Footer #2"), make sure both sidebars have some widgets (you can manage widgets here: /wp-admin/widgets.php)
- Switch to a theme that has less sidebars, say Twenty Twenty-One which has only one sidebar ("Footer").
- Ensure that the Jetpack Search sidebar has the same widgets before the theme switch, meaning it doesn't have a bunch of unwanted widgets in it, like 'Archives', 'Categories' and 'Meta' etc.
